waspc/src/Wasp/Util/WebRouterPath.hs

```diff
@@ -1,18 +1,39 @@
-module Wasp.Util.WebRouterPath where
+module Wasp.Util.WebRouterPath
+  ( Segment (StaticSegment, ParamSegment),
+    StaticSegment (RequiredStaticSegment, OptionalStaticSegment),
+    ParamSegment (RequiredParamSegment, OptionalParamSegment),
+    getRouteSegments,
+  )
+where
 
 import Data.List (isSuffixOf)
 import Data.List.Split (splitOn)
-import Data.Maybe (mapMaybe)
 
-data Param = Optional String | Required String deriving (Show, Eq)
+data Segment = StaticSegment StaticSegment | ParamSegment ParamSegment deriving (Show, Eq)
 
-extractPathParams :: String -> [Param]
-extractPathParams = mapMaybe parseParam . splitOn "/"
+data StaticSegment = RequiredStaticSegment StaticSegmentValue | OptionalStaticSegment StaticSegmentValue deriving (Show, Eq)
+
+data ParamSegment = RequiredParamSegment ParamName | OptionalParamSegment ParamName deriving (Show, Eq)
```

Comment:
I recommend breaking these up into multiple lines, more readable that way. Ormolu won't do it unless you nudge it (and it's unfortunatley not configurable).

Reply:
The `Segment` is still in one line. Not sure if that's on purpose.

Reply:
Ah yes, my bad! Changed now :)